substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateOptions.java

```diff
@@ -470,13 +472,13 @@ protected void onValueUpdate(EconomicMap<OptionKey<?>, Object> values, Integer o
     public static final HostedOptionKey<String[]> DebugInfoSourceSearchPath = new HostedOptionKey<String[]>(null) {
     };
     @Option(help = "Directory under which to create source file cache for Application or GraalVM classes")//
-    public static final HostedOptionKey<String> DebugInfoSourceCacheRoot = new HostedOptionKey<String>("sources") {
-        @Override
-        protected void onValueUpdate(EconomicMap<OptionKey<?>, Object> values, String oldValue, String newValue) {
-            // disallow null or empty string
-            if (newValue == null || newValue.length() == 0) {
-                DebugInfoSourceCacheRoot.update(values, oldValue);
-            }
+    public static final HostedOptionKey<String> DebugInfoSourceCacheRoot = new HostedOptionKey<String>("sources");
+
+    public static String getDebugInfoSourceCacheRoot() {
```

Comment:
Now that you have that abstraction in-between you could take the checking one level further by returning a `Path` here and react on `InvalidPathException` during construction with a sensible `UserError.abort` message.

Reply:
done